src/xmipp/libraries/classification/vector_ops.h

```diff
@@ -379,7 +380,8 @@ double euclideanDistance(const std::vector<T>& _v1, const std::vector<T>& _v2)
         throw std::runtime_error("vector of different size in eDist");
 
     double dist = 0;
-    typename std::vector<T>::const_iterator i, j;
+    typename std::vector<T>::const_iterator i;
+    typename std::vector<T>::const_iterator j;
     for (i = _v1.begin(), j = _v2.begin() ; i < _v1.end(); i++, j++)
```

Comment:
use `auto`

Reply:
It seems changing the `typename` to `auto` leads to a compilation error.